discovery/src/main/java/com/ibm/watson/developer_cloud/discovery/v1/model/Collection.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright 2017 IBM Corp. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package com.ibm.watson.developer_cloud.discovery.v1.model;
+
+import com.google.gson.annotations.SerializedName;
+import com.ibm.watson.developer_cloud.service.model.GenericModel;
+
+import java.util.Date;
+
+/**
+ * A collection for storing documents.
+ */
+public class Collection extends GenericModel {
+
+  /**
+   * The status of the collection.
+   */
+  public interface Status {
```

Comment:
Handling of enums is still open for discussion, but I chose to use interface in this pass because it allows developers to used defined values rather than strings but does not constrain the field to a fixed/never-changing set of values.  This seems like a good compromise that helps developers while allowing for forward compatibility.